build.rs

```diff
@@ -12,14 +12,23 @@
 // limitations under the License.
 
 use regex::Regex;
+use std::env;
+use std::fs::read_dir;
 use std::fs::File;
-use std::fs::{read_dir, remove_file};
 use std::io::{Read, Write};
 use std::process::Command;
 
 fn main() {
+    // This build script creates files in the `src` directory. Since that is
+    // outside Cargo's OUT_DIR it will cause an error when this crate is used
+    // as a dependency. Therefore, the user must opt-in to regenerating the
+    // Rust files.
+    if env::var_os("CARGO_FEATURE_REGENERATE").is_none() {
```

Comment:
Maybe `cargo:rerun-if-env-changed` is necessary otherwise the change to the variable will not trigger rebuild.

Reply:
It is a Cargo feature, not an env var, so I don't think it is necessary

Reply:
Oh, why not use `cfg!(feature = "regenerate")`, which seems more obvious.

Reply:
that doesn't work in build scripts, only in the main Rust files

Reply:
Why? `cfg!` should work just like `env::var_os` because the features used when compiling the build script should always be the same as when running it.

Reply:
You're right! I didn't know that worked